File: program/viewer.py
#!/data/salomonis2/LabFiles/Frank-Li/citeseq/scanpy_new_env/bin/python3.6

from yattag import Doc
import json


def html_banner():
    doc,tag,text,line = Doc().ttl()
    with tag('div'):
        doc.attr(klass='banner')
        with tag('h2'):
            text('scTriangulate Viewer')
    return doc.getvalue()


def html_left_nav(key_cluster_dict):
    doc,tag,text,line = Doc().ttl()
    with tag('div'):
        doc.attr(klass='left_nav')
        for key,value in key_cluster_dict.items():
            with tag('h3'):
                text(key)
            with tag('ul'):
                doc.attr(klass='{}'.format(key))
                for cluster in value:
                    with tag('li'):
                        doc.attr(onclick='display_score(this.parentElement.className,this.textContent);display_plot(this.parentElement.className,this.textContent)')
                        text(cluster)
    return doc.getvalue()

def html_right_show(key_cluster_data):
    doc,tag,text,line = Doc().ttl()
    with tag('div'):
        doc.attr(klass='right_show')
        with tag('div'):
            doc.attr(klass='score_information')
            with tag('h2'):
                doc.attr(klass='score_h2')
                text('placeholder')
            with tag('p'):
                doc.attr(klass='score_p1')
                text('placeholder')
            with tag('p'):
                doc.attr(klass='score_p2')
                text('placeholder')   
            with tag('p'):
                doc.attr(klass='score_p3')
                text('placeholder')  
        with tag('div'):
            doc.attr(klass='doublet')
            with tag('h2'):
                doc.attr(klass='doublet_h2')
                text('placeholder')
            doc.stag('img',src='./doublet.png',width='60%')

        with tag('div'):
            doc.attr(klass='enrichment')
            with tag('h2'):
                text('enrichment plot')
            doc.stag('img',src='./init.png',alt='Choose key and cluster',klass='img_enrichment',width='60%')
        with tag('div'):
            doc.attr(klass='marker_umap')
            with tag('h2'):
                text('marker gene umap')
            doc.stag('img',src='./init.png',alt='Choose key and cluster',klass='img_marker',width='90%')
        with tag('div'):
            doc.attr(klass='exclusive_umap')
            with tag('h2'):
                text('exclusive gene umap')
            doc.stag('img',src='./init.png',alt='Choose key and cluster',klass='img_exclusive',width='90%')

        with tag('div'):
            doc.attr(klass='pass_to_js')
            with tag('p'):
                text(json.dumps(key_cluster_data))
    return doc.getvalue()


def to_html(key_cluster_dict,key_cluster_data):
    doc,tag,text,line = Doc().ttl()
    doc.asis('<!DOCTYPE html>')
    with tag('html'):
        with tag('head'):
            with tag('title'):
                text('scTriangulate')
            with tag('link'):
                doc.attr(rel='stylesheet',href='./viewer.css')
        with tag('body'):
            doc.asis(html_banner())
            with tag('div'):
                doc.attr(klass='main_body')
                doc.asis(html_left_nav(key_cluster_dict))
                doc.asis(html_right_show(key_cluster_data))
        with tag('script'):
            doc.attr(src='../score.json')
            doc.attr(src='./viewer.js')
    return doc.getvalue()

File: update/sctriangulate_tfidf1/viewer/viewer.js
"use strict";


function display_score(key,cluster) {
    // get score
    var score = document.getElementsByClassName('pass_to_js')[0].getElementsByTagName('p')[0].innerText;
    var data_from_json = JSON.parse(score);
    console.log(data_from_json);
    var display_h2_span = `${key}, ${cluster}:`;
    var display_reassign = `${data_from_json[key][0][cluster]}`;
    var display_tfidf = `${data_from_json[key][1][cluster]}`;
    var display_sccaf = `${data_from_json[key][2][cluster]}`;
    var doublet = `Average doublet score is: ${data_from_json[key][3][cluster]}`;

    var h2_obj = document.getElementById('score_information').getElementsByTagName('span');
    h2_obj[0].innerText = display_h2_span;
    var reassign = document.getElementById('reassign');
    var tfidf = document.getElementById('tfidf');
    var sccaf = document.getElementById('sccaf');
    reassign.innerText = display_reassign;
    tfidf.innerText = display_tfidf;
    sccaf.innerText = display_sccaf;
    var doublet_obj = document.getElementById('doublet').getElementsByTagName('h2');
    doublet_obj[0].innerText = doublet;

}

function display_plot(key,cluster) {

    var path = `./${key}_${cluster}_location_umap.png`;
    var img_identity = document.getElementsByClassName('img_identity');
    img_identity[0].src = path;

    var path = `./${key}_${cluster}_enrichment.png`;
    var img_enrichment = document.getElementsByClassName('img_enrichment');
    img_enrichment[0].src = path;

    var path = `./${key}_${cluster}_marker_umap.png`;
    var img_marker = document.getElementsByClassName('img_marker');
    img_marker[0].src = path;

    var path = `./${key}_${cluster}_exclusive_umap.png`;
    var img_marker = document.getElementsByClassName('img_exclusive');
    img_marker[0].src = path;
}




